### PR TITLE
[MERGE] Adding support for cert-manager certificates to manage webhook TLS

### PR DESCRIPTION
**Pull request by YanivKunda**
This PR combines changes from https://github.com/kubeflow/spark-operator/pull/1567 and  https://github.com/kubeflow/spark-operator/pull/2079, with the following content:

1. Merging the changes from [dskatz:feature/certmanager](https://github.com/dskatz/spark-on-k8s-operator/tree/feature/certmanager)
2. Resolving conflicts (keeping `webhook-cleanup-job.yaml` & `webhook-init-job.yaml` deleted)
3. Upgrading `cert-manager.io` apiVersion to v1
4. Adding additional suggestions from https://github.com/kubeflow/spark-operator/pull/2079
5. Cleaning merge mess and tidying trailers
6. Bumping operator app and chart versions

This should resolve issue https://github.com/kubeflow/spark-operator/issues/1178

### CODE REVIEW

**Comment by YanivKunda** [comment]
@yuchaoran2011 / @ChenYi015 - I moved over the changes from https://github.com/kubeflow/spark-operator/pull/2079 here due to merge and trailers issues, let me know what else is required.

**Comment by YanivKunda** [comment]
@yuchaoran2011  / @ChenYi015 - Anything that needs to be done on my side?

**Comment by yuchaoran2011** [comment]
@YanivKunda Missed this one earlier. Looks like another set of conflicts has happened. could you fix them?

File: charts/spark-operator-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -101,6 +101,11 @@ spec:
         - -webhook-timeout={{ .Values.webhook.timeout }}
         - -webhook-namespace-selector={{ .Values.webhook.namespaceSelector }}
         - -webhook-object-selector={{ .Values.webhook.objectSelector }}
+          {{- if .Values.webhook.certManager.enable }}
```
Comment (Cian911) [inline charts/spark-operator-chart/templates/deployment.yaml:104]: I don't think this is going to work, as these flags were removed in a previous [PR](https://github.com/kubeflow/spark-operator/pull/2016/files#diff-df64e37f2222b761c62bd8bf32c0347aae0fe1f46ff6acf0e66c2b61bd30d761L107-L112) by @ChenYi015 

However, there is value in being able to provide our own certs, perhaps they should be added back @ChenYi015

Comment (ChenYi015) [reply charts/spark-operator-chart/templates/deployment.yaml:104]: @Cian911  Make sense, I will do it.

Comment (YanivKunda) [reply charts/spark-operator-chart/templates/deployment.yaml]: @ChenYi015 I've accidentally discarded the changes in my fork -
but trying to re-apply them now looks a bit more complicated after the big controller-runtime restructuring.
What are your thoughts?

**Pull request closed without merging**